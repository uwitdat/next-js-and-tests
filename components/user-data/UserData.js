import React, { useState } from 'react'
import { useRouter } from 'next/router';
import AddUserForm from '../add-user-form/AddUserForm';
import Card from '@mui/material/Card';
import CardActions from '@mui/material/CardActions';
import CardContent from '@mui/material/CardContent';
import Image from 'next/image';
import Button from '@mui/material/Button';
import Typography from '@mui/material/Typography';
import userStyles from './user-styles.module.css';


const UserData = ({ data }) => {
  const [users, setUsers] = useState(data);
  const router = useRouter();

  const handleGoToProfile = (user) => {
    router.push(
      {
        pathname: `/users/${user.id}`,
        query: { user: JSON.stringify(user) },
      },
      `/users/${user.id}`,
    )
  }

  return (
    <section className={userStyles.container}>
      <h1>Users:</h1>
      {users.map((user) => (
        <Card sx={{ maxWidth: 345, width: '70%', margin: '1rem', display: 'flex', flexDirection: 'column', alignItems: 'center' }} key={user.id}>
          <Image
            width={100}
            height={100}
            src={'https://www.kindpng.com/picc/m/24-248253_user-profile-default-image-png-clipart-png-download.png'} />
          <CardContent>
            <Typography gutterBottom variant="h5" component="div">
              {user.name}
            </Typography>
            <Typography variant="body2" color="text.secondary">
              {user.email}
            </Typography>
          </CardContent>
          <CardActions>
            <Button onClick={() => handleGoToProfile(user)} size="small">View Profile</Button>
          </CardActions>
        </Card>
      ))}
      <AddUserForm users={users} setUsers={setUsers} />
    </section>
  )
}

export default UserData

