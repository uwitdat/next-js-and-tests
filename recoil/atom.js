import { atom } from 'recoil';

export const usersState = atom({
    key: 'usersState',
    default: '', // default value (aka initial value)
});