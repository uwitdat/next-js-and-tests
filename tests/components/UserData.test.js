import React from 'react';
import { render, cleanup, } from "@testing-library/react";
import '@testing-library/jest-dom';
import UserData from '../../components/user-data/UserData';
import renderer from 'react-test-renderer';
import { RecoilRoot } from 'recoil';

afterEach(() => {
  cleanup();
});

const users = [
  {
    id: 1,
    name: 'ben',
    email: "email@email.com",
    phone: '111-111-1111',
    website: 'some site'
  },
  {
    id: 2,
    name: 'ben two',
    email: "emailtwo@email.com",
    phone: '222-222-2222',
    website: 'some site two'
  },
];

test('<UserData/> and its props render without crashing', () => {
  const { getByTestId, getByText } = render(
    <RecoilRoot>
      <UserData users={users} />
    </RecoilRoot>
  );

  users.forEach((user) => {
    const imageComponent = getByTestId(`user-img-${user.id}`);

    expect(getByText(user.name)).toBeVisible();
    expect(getByText(user.email)).toBeVisible();

    expect(imageComponent).toBeInTheDocument();
  });

});


test('<UserData/> matches snapshot', () => {
  const tree = renderer.create(
    <RecoilRoot>
      <UserData data={users} />
    </RecoilRoot>
  ).toJSON();
  expect(tree).toMatchSnapshot();
})