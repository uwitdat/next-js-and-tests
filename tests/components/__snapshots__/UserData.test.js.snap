// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserData/> matches snapshot 1`] = `
<p>
  loading...
</p>
`;
